### PR TITLE
Support rt record functions

### PR DESCRIPTION
**Pull request by RolandMacDoland**
This PR addresses the support for untagged runtime functions namely:

- `integer_record_output`
- `bool_record_output`
- `result_record_output`

and support for `read_result` instructions and few other bits like passing constant variables to functions.

### CODE REVIEW

**Comment by cqc-alec** [comment]
Does this require the binary wheel you have added in this PR?

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -35,4 +35,4 @@ jobs:
       - name: Run tests
         working-directory: ./tests
         run: |
-          poetry run pytest qir_test.py
+          poetry run pytest parser_test.py generator_test.py
```
Comment (cqc-alec) [inline .github/workflows/test.yml:38]: Have all the tests for `qir_test.py` moved into `parser_test.py` and `generator_test.py`? If so, `qir_test.py` should be deleted; if not, it should be run!

File: pytket_qir/gatesets/base.py [hunk 2]
```diff
@@ -38,13 +40,18 @@ class OpName(Enum):
     AND = "and"
     OR = "or"
     XOR = "xor"
+    INT = "integer"
+    BOOL = "bool"
+    RES = "result"
```
Comment (cqc-alec) [inline pytket_qir/gatesets/base.py:45]: These look like types rather than ops?

File: tests/conftest.py [hunk 3]
```diff
@@ -19,6 +19,7 @@
 
 from pytest import fixture  # type: ignore
 
+# from pyqir import Builder, IntPredicate, Value  # type: ignore
```
Comment (cqc-alec) [inline tests/conftest.py]: Remove commented line.

File: .github/workflows/test.yml (hunk 1 above)
Comment (cqc-alec) [reply .github/workflows/test.yml:38]: Ah, I see it has been renamed, nm.

**Review by cqc-alec — CHANGES_REQUESTED** [review]
Generally seems good, question about the wheel.

**Comment by RolandMacDoland** [comment]
> Does this require the binary wheel you have added in this PR?

No, that's right. Removed. Thanks.

File: tests/conftest.py (hunk 3 above)
Comment (RolandMacDoland) [reply tests/conftest.py]: Thanks. Done.

File: pytket_qir/gatesets/base.py (hunk 2 above)
Comment (RolandMacDoland) [reply pytket_qir/gatesets/base.py:45]: That's right. Maybe I should rename these datatypes something like:: `FuncName`, `FuncSpec` etc ?

Comment (RolandMacDoland) [reply pytket_qir/gatesets/base.py:45]: Done.

**Review by cqc-alec — APPROVED** [review]

**Pull request merged**